spec/requests/external_users/certification_spec.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+RSpec.describe 'Certification', type: :request do
+  let(:advocate) { create(:external_user, :advocate) }
+  let(:claim) { create(:advocate_final_claim) }
+
+  describe 'POST #create' do
+    let(:certification_params) do
+      {
+        'claim_id' => claim.id,
+        'commit' => 'Certify and Submit Claim',
+        'certification' => {
+          'certification_type_id' => '1',
+          'certified_by' => 'Joe Bloggs',
+          'main_hearing' => 'true',
+          'certification_date(3i)' => '31',
+          'certification_date(2i)' => '01',
+          'certification_date(1i)' => '2021'
+        }
+      }
+    end
+
+    context 'when user is not signed in' do
+      before do
+        post external_users_claim_certification_path(claim.id), params: certification_params
+      end
+
+      it 'redirects to sign in page' do
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when user is signed in' do
+      before do
+        sign_in advocate.user
+      end
+
+      context 'with invalid parameters' do
+        let(:certification_params) do
+          {
+            'claim_id' => claim.id,
+            'commit' => 'Certify and Submit Claim',
+            'certification' => {
+              'certification_type_id' => '1',
+              'certified_by' => 'Joe Bloggs',
+              'main_hearing' => 'true',
+              'certification_date(3i)' => '32',
+              'certification_date(2i)' => 'JAN',
+              'certification_date(1i)' => '2021'
+            }
+          }
+        end
+
+        it 'invalid multiparameter date parts do not raise ActiveRecord::MultiparameterAssignmentErrors' do
+          post external_users_claim_certification_path(claim.id), params: certification_params
+          expect(response).to render_template(:new)
+        end
```

Comment:
For a test with this name I would expect something like;

```ruby
        it 'invalid multiparameter date parts do not raise ActiveRecord::MultiparameterAssignmentErrors' do
          expect { post external_users_claim_certification_path(claim.id), params: certification_params }
            .not_to raise_error(ActiveRecord::MultiparameterAssignmentErrors)
        end
```

Would it be better named something like "renders the new certification template" and have another test for the correct status code?

Reply:
yes, but generally better to have positive tests to avoid false positives... and rubocop says
```
WARNING: Using `expect { }.not_to raise_error(SpecificErrorClass)` risks false positives,
```
will tweak anyway

Reply:
I would just change the name of the test. What is important is that the page is rendered correctly, not that an exception is not raised.